solutions/gold/cses-1715.mdx

```diff
@@ -66,4 +70,56 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+
+```py
+from collections import Counter
+
+MAXN = 10**6
+MOD = 10**9 + 7
+
+fac = [0] * (MAXN + 1)
+inv = [0] * (MAXN + 1)
+
+
+# BeginCodeSnip{Combinatorics Functions (from the module)}
+def exp(x: int, n: int, m: int) -> int:
+	x %= m
+	res = 1
+	while n > 0:
+		if n % 2 == 1:
+			res = res * x % m
+		x = x * x % m
+		n //= 2
+	return res
+
+
+def factorial():
+	fac[0] = 1
+	for i in range(1, MAXN + 1):
+		fac[i] = fac[i - 1] * i % MOD
+
+
+def inverses():
+	inv[MAXN] = exp(fac[MAXN], MOD - 2, MOD)
+	for i in range(MAXN, 0, -1):
+		inv[i - 1] = inv[i] * i % MOD
+
+
+# EndCodeSnip
+
+st = input()
+cnt = Counter(st)
+factorial()
```

Comment:
did you really need to make a separate method for those lol

Reply:
The template was already used in several solutions so ig why not use it rather than something new

Reply:
oh ic it is in the module, fair enough